xmldb/database.go

```diff
@@ -0,0 +1,121 @@
+package xmldb
+
+import (
+	"errors"
+	"fmt"
+	"github.com/Project-Planner/backend/model"
+	"github.com/google/uuid"
+	"log"
+	"os"
+	"path/filepath"
+	"regexp"
+	"strings"
+)
+
+// The struct implementing the web.Database interface
+type Database struct {
+	config DBConfig
+	auth   map[string]model.Login
+	users  map[string]model.User
+}
+
+func NewDatabase(config DBConfig) (Database, error) {
+	// 1. Step: Ensuring that the parent folders (auth, user, calendars) exist.
+	ensureDir(config.RootDir)
+	ensureDir(config.AuthDir)
+	ensureDir(config.UserDir)
+	ensureDir(config.CalendarDir)
+
+	//2. Step: Parsing source files into corresponding structs or struct collections.
+	// #Auth#
+	// a) Ensuring that the authentication file actually exists.
+	//	  If it doesn't, then it is created, so that parsing
+	//	  can go on.
+	var authPath = fmt.Sprintf("%s/%s", config.AuthDir, "/auth.xml")
+	var authPayload model.Auth
+	var auth = make(map[string]model.Login)
```

Comment:
In go würde man das eher so schreiben. Kenne niemand der das so mit `var name =` macht. Aber kannst auch gern so lassen. Ist personel preference, aber gibt mir immer so JS vibes. Wollte es nur gesagt haben 
```suggestion
	auth := make(map[string]model.Login)
```